Update usage instructions for Flatpak

guihkx opened the pull request:
This simplifies things regarding the Flatpak set up. This also adds:

- Flatpak-specific build instructions.
- Remove inaccurate comment about the Flatpak package.
- Remove every `$` at the beginning of each command for easier copying and pasting.

Rendered page: https://github.com/guihkx/spotify-adblock/blob/readme-flatpak/README.md

Closes #112

guihkx commented:
Apologies for the extra force-pushes. It should be ready for review now.

axelsimon commented:
@guihkx small mistake in your instructions.

`mkdir -p ~/.var/app/com.spotify.Client/config/spotify-adblock`
but then:
`cp config.toml ~/.var/app/com.spotify.Client/spotify-adblock` 

`/config` is missing, the right command is:
`cp config.toml ~/.var/app/com.spotify.Client/config/spotify-adblock` 

Also, i'm not sure i like telling people they just need to use podman to build. At least we should give people the option: either you need cargo + rust, or if you don't want to install that you can run a container using podman (why Ubuntu 23.10 though, at least use a LTS or a recent version). But then again you can also use nix and `nix-shell -p cargo` so, i'm not sure we should push flatpak users to a specific solution.

axelsimon reviewed (commented):
Needed change.

guihkx commented on README.md:
```diff
@@ -14,32 +13,43 @@ Prerequisites:
-$ LD_PRELOAD=/usr/local/lib/spotify-adblock.so spotify
+mkdir -p ~/.var/app/com.spotify.Client/config/spotify-adblock
+cp target/release/libspotifyadblock.so ~/.var/app/com.spotify.Client
+cp config.toml ~/.var/app/com.spotify.Client/spotify-adblock
```
Good catch, thanks.

guihkx commented:
Updates:

- Fixed the `cp config.toml ...` command not targeting the right location.
- Updated the Flatpak-specific build instructions to use a Ubuntu 24.10 container image, since it has [glibc 2.40](https://packages.ubuntu.com/oracular/libc6), which matches the version used by the runtime Spotify for Flatpak currently runs on ([24.08](https://github.com/flathub/com.spotify.Client/blob/519ed65e99cb3541d9ed3b217afa1c107c00b356/com.spotify.Client.json#L3-L4)).

guihkx commented:
>I think telling people to build the project using podman/docker adds a big dependency, complexity and potential issues from first-time users trying to set it up in their environment.

Even though I disagree, I've now removed the build instructions for Flatpak.

Also, [as stated in the README](https://github.com/abba23/spotify-adblock/blob/fc0d2e21d886f5dbb9dc1f20ad51daaddd2cf737/README.md?plain=1#L6), if they use the `.so` library you provide, or even if they build manually using the host's Rust compiler, it *might* not work with the Flatpak runtime due to potentially different glibc versions (see #24 and #28).

This would be solved by building the library with Podman/Docker using a container that has the same glibc version used by the Flatpak runtime used by the Spotify app.

> Since I don't use Flatpak myself, I can't verify, maintain or help people having problems with these instructions either.

I guarantee they work, but I also understand what you're saying. We'd also have to update the build instructions for Flatpak yearly (i.e. when they move to a newer runtime, which is released once a year).

> Either way I'm not convinced that removing the `$`s is a good idea, because the easier copying comes at the cost of readability

Alright, I've added them back.
